src/index.ts

```diff
@@ -285,4 +285,35 @@ export default class Collection<Resource extends { [key: string]: any }> {
       success: value => RD.success(union(value, [id]))
     });
   }
+
+  public toJSON(): {
+    _URI: URI;
+    knownIds: RemoteList<string>;
+    idMap: ById<RemoteList<string>>;
+    entities: ById<Remote<Resource>>;
+  } {
+    return {
+      _URI: URI,
+      knownIds: this.knownIds,
+      idMap: this.idMap.value,
+      entities: this.entities
+    };
+  }
+}
+
+export function fromJSON<A>(_: string, value: any): Collection<A> {
+  if (value && value._URI === URI) {
+    const remoteCollection: Collection<A> = new Collection<A>();
+    remoteCollection.entities = new StrMap<Remote<A>>(value.entities).map((entity: Remote<A>) =>
+      fromRemoteJSON(entity)
```

Comment:
nit: Instead of re-importing the export, you could just `RD.fromJSON`.

Reply:
actually, not a nit, this _has_ to be `import * as RD` since a bunch of other bits are using `RD.whatevs`. Tests are failing as-is.